pki/deploy.go

```diff
@@ -61,6 +61,29 @@ func DeployCertificatesOnWorkers(workerHosts []*hosts.Host, crtMap map[string]Ce
 	return nil
 }
 
+func DeployCertificatesOnEtcd(etcdHosts []*hosts.Host, crtMap map[string]CertificatePKI, certDownloaderImage string) error {
+	// list of certificates that should be deployed on the etcd
+	crtList := []string{
+		CACertName,
+	}
+	for i := range etcdHosts {
+		crtList = append(crtList, fmt.Sprintf("%s-%d", EtcdName, i))
+	}
+	env := []string{}
+	for _, crtName := range crtList {
+		c := crtMap[crtName]
```

Comment:
here we expect the key in crdMap be the same as crt name we generate on the line 70. Consider moving the generation code to its separate method so you don't have to maintain it in 2 places

Reply:
Also is it possible that the cert `crtName` does not exist in map? If yes, we should handle it